sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -286,6 +286,12 @@ class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan]
         _, _, _, _, _, _) =>
       i.copy(table = DDLUtils.readHiveTable(tableMeta))
 
+    case data @ AppendData(
+        DataSourceV2Relation(V1Table(table: CatalogTable), _, _, _, _), _, _, _, _, _) =>
+      InsertIntoStatement(UnresolvedCatalogRelation(table),
+        table.partitionColumnNames.map(name => name -> None).toMap,
+        Seq.empty, data.query, false, false)
```

Comment:
we should get `byName` from `AppendData`

Reply:
actually we should only do this conversion if `byName` is false

Reply:
ah done